soc/nordic/nrf54h/soc.h

```diff
@@ -36,4 +37,14 @@
 #define RAMBLOCK_RET2_BIT_DCACHE MEMCONF_POWER_RET2_MEM7_Pos
 #endif
 
+/**
+ * @brief Get the ID of the node used by the power domain management.
+ *
+ * This function returns the node ID to be shared with other dependent modules.
+ *
+ * @return The pointer to the node assigned to the SOC module
+ *	   used for the power domain management purposes.
+ */
+sys_snode_t *soc_pd_sys_snode_get(void);
```

Comment:
This looks quite weird. Why not just use `extern sys_snode_t soc_node;` in `power.c`, for example? Is this node intended to be used by any other modules/subsystems?